R/lasso-deployment.R

```diff
@@ -219,68 +293,35 @@ LassoDeployment <- R6Class(
       RODBC::odbcCloseAll()
     },
 
-    fitGeneralizedLinearModel = function() {
-      if (isTRUE(self$params$debug)) {
-        print("generating fitLogit...")
-      }
-
-      if (self$params$type == "classification") {
-        private$fitLogit <- glm(as.formula(paste(self$params$predictedCol, ".", sep = " ~ ")),
-                                data = private$dfTrain, family = binomial(link = "logit"), metric = "ROC",
-                                control = list(maxit = 10000), trControl = trainControl(classProbs = TRUE,
-                                                                                        summaryFunction = twoClassSummary))
-
-      } else if (self$params$type == "regression") {
-        private$fitLogit <- glm(as.formula(paste(self$params$predictedCol, ".", sep = " ~ ")),
-                                data = private$dfTrain, metric = "RMSE", control = list(maxit = 10000))
-      }
-    },
-
-    saveModel = function() {
-      if (isTRUE(self$params$debug)) {
-        print("Saving model...")
-      }
-
-      # NOTE: save(private$fit, ...) does not work!
-      if (isTRUE(!self$params$useSavedModel)) {
-        fitObj <- private$fit
-        save(fitObj, file = "rmodel_combined.rda")
-      }
-
-      # This isn't needed if formula interface is used in randomForest
-      private$dfTest[[self$params$predictedCol]] = NULL
-
-      if (isTRUE(self$params$debug)) {
-        print("Test set before being used in predict(), after removing y")
-        print(str(private$dfTest))
-      }
-    },
-
+    # Predict results
     performPrediction = function() {
-      if (self$params$type == "classification") {
-        # linear , these are probabilities
-        private$predictedVals <- predict(private$fit, newdata = private$dfTest, type = "response")
-        private$predictedValsForUnitTest <- private$predictedVals[5]  # for unit test
-
-        print("Probability predictions are based on logistic")
-
-        if (isTRUE(self$params$debug)) {
-          print(paste0("Rows in prob prediction: ", nrow(private$predictedVals)))
+      # Index of largest lambda within one cvse of the lambda with lowest cve:
+      # These are sorted from largest to smallest lambda, hence pulling the
+      # minimum index.
+      private$indLambda1se <- min(which(private$fitGrLasso$cve <= (private$fitGrLasso$cve + private$fitGrLasso$cvse)[private$fitGrLasso$min]))
```

Comment:
Do you mind taking this to under 80 chars?

Also, please see if it works without the `which`. It has been known to swallow errors...